packages/react/runtime/lepus/index.js

```diff
@@ -41,3 +41,47 @@ export function createElement(type, props, children) {
 
   return createVNode(type, normalizedProps, key, ref, null);
 }
+
+/**
+ * Clones the given VNode, optionally adding attributes/props and replacing its
+ * children.
+ * @param {VNode} vnode The virtual DOM element to clone
+ * @param {object} props Attributes/props to add when cloning
+ * @param {Array<ComponentChildren>} rest Any additional arguments will be used
+ * as replacement children.
+ * @returns {VNode}
+ */
+export function cloneElement(vnode, props, children) {
```

Comment:
We may reduce some bytes to use `createElement` in `cloneElement`:

```js
export function cloneElement(vnode, props, ...rest) {
	return createElement(
		vnode.type,
		assign(
			vnode.ref ? { ref: vnode.ref } : {},
			vnode.key ? { key: vnode.key } : {},
			vnode.props,
			props
		),
		...rest
	);
}
```

> https://github.com/preactjs/preact/pull/4814/commits/37d531f55078bb5b37837baf3561dee7947a1f8f

> [!NOTE]
> Preact did not merge this change since it would increase the **compressed size**.
> But I think it would be OK for us since we are using AOT bytecode in main-thread.